lib/archethic/p2p/message/validate_smart_contract_call.ex

```diff
@@ -78,7 +80,8 @@ defmodule Archethic.P2P.Message.ValidateSmartContractCall do
 
     with {:ok, contract_tx} <- TransactionChain.get_transaction(recipient_address),
          {:ok, contract} <- Contracts.from_transaction(contract_tx),
-         trigger when not is_nil(trigger) <- Contract.get_trigger_for_recipient(recipient),
+         {:trigger, trigger} when not is_nil(trigger) <-
```

Comment:
Maybe you should add a private function to continue to leverage `{:ok, _}` syntax. 
```elixir
defp validate_trigger(recipient) do
  case Contract.get_trigger_for_recipient(recipient) do
    nil ->
      {:error, :invalid_trigger}
    _ ->
      {:ok, trigger}
  end
end
```
This would make the code more readable and consistent